src/host/exemain.cpp

```diff
@@ -83,9 +83,9 @@ static bool ShouldUseConhostV2()
     }
 }
 
-static bool ShouldUseLegacyConhost(const bool fForceV1)
+static bool ShouldUseLegacyConhost(const ConsoleArguments& args)
 {
-    return fForceV1 || !ShouldUseConhostV2();
+    return (args.GetForceV1() || !ShouldUseConhostV2()) && !args.InConptyMode();

```

Comment:
OK this is what we need, but can you make this a little less mentally stressful to read like
```
if (args.InConptyMode())
{
     return false;
}
// etc...
```

Also, if you think `ShouldUseConhostV2` is poorly named, please just rename it right now to be `DoesRegistryPrefSayUseConhostV2` or something to clarify it's just the registry lookup and not something more elaborate than that.

Reply:
I hate that the most ideal name for that function is `IsConhostV2NotForcedOffInRegistry` (because it's _unset or nonzero = v2_, but set to 0 = v1)

Reply:
`DoesRegistryForceConhostV1`?

Reply:
That's not unreasonable. I want to keep the change scoped down so that it's more easily serviceable, though, and that requires us to flip the sense on the function

Reply:
Admittedly, it is clearer.